frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/pipeline-details/PipelineDetails.tsx

```diff
@@ -3,24 +3,34 @@ import { SectionHeading, ResourceSummary, ResourceLink } from '@console/internal
 import { referenceForModel } from '@console/internal/module/k8s';
 import { Pipeline, getResourceModelFromTask } from '../../../../utils/pipeline-augment';
 import PipelineVisualization from './PipelineVisualization';
+import { usePipelineTriggerTemplateNames } from '../../utils/triggers';
+import { TriggerTemplateModel } from '../../../../models';
+import ResourceLinkList from '../../resource-overview/ResourceLinkList';
 
 interface PipelineDetailsProps {
   obj: Pipeline;
 }
 
-const PipelineDetails: React.FC<PipelineDetailsProps> = ({ obj: pipeline }) => (
-  <div className="co-m-pane__body">
-    <SectionHeading text="Pipeline Details" />
-    <PipelineVisualization pipeline={pipeline} />
-    <div className="row">
-      <div className="col-sm-6">
-        <ResourceSummary resource={pipeline} />
-      </div>
-      {pipeline.spec && pipeline.spec.tasks && (
+const PipelineDetails: React.FC<PipelineDetailsProps> = ({ obj: pipeline }) => {
+  const triggerTemplateNames: string[] = usePipelineTriggerTemplateNames(pipeline) || [];
+  return (
+    <div className="co-m-pane__body">
+      <SectionHeading text="Pipeline Details" />
+      <PipelineVisualization pipeline={pipeline} />
+      <div className="row">
+        <div className="col-sm-6">
+          <ResourceSummary resource={pipeline} />
+        </div>
         <div className="col-sm-6">
+          <ResourceLinkList
+            title={TriggerTemplateModel.labelPlural}
+            namespace={pipeline.metadata.namespace}
+            kind={referenceForModel(TriggerTemplateModel)}
+            links={triggerTemplateNames}
+          />
           <SectionHeading text="Tasks" />
           <dl>
-            {pipeline.spec.tasks.map((task) => {
+            {(pipeline.spec?.tasks || []).map((task) => {
```

Comment:
Is there a reason you think this is needed?

```suggestion
            {pipeline.spec.tasks.map((task) => {
```

Can we not just update Pipeline to not have an [optional spec](https://github.com/openshift/console/blob/fc4406944167922acde1c71c4ed4c29958eebbcc/frontend/packages/dev-console/src/utils/pipeline-augment.ts#L84)?

Reply:
updated